controllers/kogitoserverlessbuild_controller.go

```diff
@@ -57,48 +59,76 @@ type KogitoServerlessBuildReconciler struct {
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.12.1/pkg/reconcile
 func (r *KogitoServerlessBuildReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	log := ctrllog.FromContext(ctx)
-	instance := &api08.KogitoServerlessBuild{}
-	err := r.Client.Get(ctx, req.NamespacedName, instance)
-	if err == nil {
-		phase := instance.Status.BuildPhase
-		if r.defaultBuildConf.Data == nil {
-			r.defaultBuildConf, err = utils.GetConfigMap(r.Client)
+	build := &api08.KogitoServerlessBuild{}
+	err := r.Client.Get(ctx, req.NamespacedName, build)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return ctrl.Result{}, nil
 		}
+		log.Error(err, "Failed to get KogitoServerlessWorkflow")
+		return ctrl.Result{}, err
+	}
+
+	phase := build.Status.BuildPhase
+	if r.commonBuildConf.Data == nil {
+		r.commonBuildConf, err = utils.GetBuilderCommonConfigMap(r.Client)
+	}
 
-		if err != nil || len(r.defaultBuildConf.Data[r.defaultBuildConf.Data[constants.DEFAULT_BUILDER_RESOURCE_NAME_KEY]]) == 0 {
-			return ctrl.Result{}, errors.NewNotFound(schema.GroupResource{
-				Resource: "ConfigMap",
-			}, "builder-config")
+	if err != nil || len(r.commonBuildConf.Data[r.commonBuildConf.Data[constants.DEFAULT_BUILDER_RESOURCE_NAME_KEY]]) == 0 {
```

Comment:
Don't need to do this here, but we should consider having an assessor for these cases. Can be very confusing to maintain.

Reply:
I removed this check and added a function to doublecheck that all the keys into the builder config map are there and not empty (the function is [here](https://github.com/davidesalerno/kogito-serverless-operator/blob/156553f1de8a30347f77c0ce10c048bb36239cd9/utils/build.go#L69))